### PR TITLE
fix(queue/merge): stop run them once it's done

### PR DESCRIPTION
**Pull request by sileht**
This change stop running queue/merge action once the pull request has
succeed onetime (got merged).

This avoid replacing the check-run "This PR has been merged...." by
"action cancelled" when people have conditions that unmatch once the PR
is merged (-merged, or automerge label removed, ...).

Fixes MRGFY-1022
Fixes MERGIFY-ENGINE-2M0

Change-Id: I72d61c9b92c6ba3dddc7489359b6e8d3ceda252d

### CODE REVIEW

File: mergify_engine/actions/merge_base.py [hunk 1]
```diff
@@ -162,10 +162,8 @@ async def _merge(
 
         if commit_title_and_message is not None:
             title, message = commit_title_and_message
-            if title:
-                data["commit_title"] = title
-            if message:
-                data["commit_message"] = message
+            data["commit_title"] = title
+            data["commit_message"] = message
```
Comment (jd) [inline mergify_engine/actions/merge_base.py:166]: unrelated?

**Review by jd — CHANGES_REQUESTED** [review]

**Comment by sileht** [comment]
@Mergifyio rebase

File: mergify_engine/actions/__init__.py [hunk 2]
```diff
@@ -63,6 +63,8 @@ class ActionFlag(enum.Flag):
     # Allow to return this command as pending if does not match its conditions
     # requirements.
     ALLOW_AS_PENDING_COMMAND = enum.auto()
+    # Once succeed the action must not be ran anymore
+    CAN_SUCCEED_ONLY_ONCE = enum.auto()
```
Comment (jd) [inline mergify_engine/actions/__init__.py:67]: I think the name is confusing
what about `SUCCESS_IS_FINAL_STATE`? or something like that.

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/actions/__init__.py (hunk 2 above)
Comment (jd) [reply mergify_engine/actions/__init__.py:67]: Or `NO_RUN_AFTER_SUCCESS`

**Review by jd — APPROVED** [review]

File: mergify_engine/actions/__init__.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/actions/__init__.py]: NO_RUN_AFTER_SUCCESS suggests we can still CANCEL, but we should not. SUCCESS_IS_FINAL_STATE looks good.
